### PR TITLE
[Toolpad Editor] Add tooltip to custom component subtitle

### LINKED ISSUE #2560: Link custom component doc from the component library
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Custom components: Add question mark behind “CUSTOM COMPONENTS” subtitle in the components editor. One line explainer in the tooltip:
> Expand Toolpad with your own React components. [Click](https://mui.com/toolpad/concepts/custom-components/) to read more.

### PR DESCRIPTION
**Pull request by rohanprasadofficial**
Fixes #2560 

Added a tooltip to the custom component title which takes the user to the docs link here : https://mui.com/toolpad/concepts/custom-components

I have added a box to make the typography and icon into one to accomodate into the higher box as one component. 

### Light mode 

https://github.com/mui/mui-toolpad/assets/30728574/047671bf-9797-47e6-9f51-dec8a3693eba

### Dark mode 

https://github.com/mui/mui-toolpad/assets/30728574/6484b27b-7b63-4fed-b9fd-f02517982998

#### Dark Mode 

dark tooltip

#### Light Mode 

light tooltip

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [x] I've updated the relevant documentation for any new or updated feature
- [x] I've linked relevant GitHub issue with "Closes #<issue id>"
- [x] I've added a visual demonstration under the form of a screenshot or video

### CODE REVIEW

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx [hunk 1]
```diff
@@ -156,7 +166,28 @@ export default function ComponentCatalog({ className }: ComponentCatalogProps) {
...
+                          target="_blank"
+                          rel="noopener"
+                        >
+                          Click{' '}
+                        </Link>
+                        to read more.
+                      </Typography>
+                    }
+                  >
+                    <HelpOutlineIcon />
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx:188]: The icon looks great, my only nitpick is that it contrasts more with the background that other more important visual elements near it. 
Could we make the color slightly more faded, both in light and dark mode? (so not the same exact color as the text)
There are more faded colors in that panel that we should be able reuse in the icon, no need to create new colors.

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Looks good, just a nitpick!
Also could we make this help icon with tooltip a more reusable component? I feel like we may want to use it in more places.

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx (hunk 1 above)
Comment (rohanprasadofficial) [reply packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx]: Update the screenshots in description and separated out component.

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx [hunk 2]
```diff
@@ -156,7 +157,26 @@ export default function ComponentCatalog({ className }: ComponentCatalogProps) {
...
+                  <HelpTooltipIcon
+                    helpText={
+                      <Typography variant="inherit">
+                        Expand Toolpad with your own React components.{' '}
+                        <Link
+                          href="https://mui.com/toolpad/concepts/custom-components"
+                          target="_blank"
+                          rel="noopener"
+                        >
+                          Click{' '}
```
Comment (oliviertassinari) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx:173]: We need to find a different link label https://developer.chrome.com/en/docs/lighthouse/seo/link-text/

Comment (rohanprasadofficial) [reply packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx:173]: @oliviertassinari maybe we can rephrase the help text to make it more descriptive with link 🤔

Comment (oliviertassinari) [reply packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx:173]: A rephrase would be great. 

It's common to see generic links, but "click" is pushing it too far, "learn more" would already bring a lot more context.

Notion

Screenshot 2023-09-03 at 00 19 00

Ashby

Screenshot 2023-09-03 at 00 25 04

Comment (rohanprasadofficial) [reply packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx]: With learn more text :

Light : 
Screenshot 2023-09-04 at 8 35 43 PM

Dark: 
Screenshot 2023-09-04 at 8 35 17 PM

File: packages/toolpad-app/src/components/HelpTooltipIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { Tooltip } from '@mui/material';
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
+
+export interface HelpTooltipIconProps {
+  helpText: React.ReactNode;
```
Comment (apedroferreira) [inline packages/toolpad-app/src/components/HelpTooltipIcon.tsx:6]: Could this be a string too? Would `string | React.ReactNode` work?

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx [hunk 4]
```diff
@@ -156,7 +157,25 @@ export default function ComponentCatalog({ className }: ComponentCatalogProps) {
...
+                  <HelpTooltipIcon
+                    helpText={
+                      <Typography variant="inherit">
+                        Expand Toolpad with your own React components.{' '}
+                        <Link
+                          href="https://mui.com/toolpad/concepts/custom-components"
+                          target="_blank"
+                          rel="noopener"
+                        >
+                          Learn more{' '}
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx:173]:
```suggestion
                          Learn more
```

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx [hunk 5]
```diff
@@ -156,7 +157,25 @@ export default function ComponentCatalog({ className }: ComponentCatalogProps) {
...
+                        </Link>
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx:174]:
```suggestion
                        </Link>.
```

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Looks great, just some very minor extra changes!

File: packages/toolpad-app/src/components/HelpTooltipIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { Tooltip } from '@mui/material';
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
+
+export interface HelpTooltipIconProps {
```
[the remark this answers is not in the document]
Comment (rohanprasadofficial) [reply packages/toolpad-app/src/components/HelpTooltipIcon.tsx:5]: good catch, otherwise it would collide with title of tooltip !

**Review by apedroferreira — APPROVED** [review]
Thanks a lot!

File: packages/toolpad-app/src/components/HelpTooltipIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { Tooltip, TooltipProps } from '@mui/material';
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
+
+export interface HelpTooltipIconProps extends Omit<TooltipProps, 'title' | 'children'> {
+  helpText: string | React.ReactNode;
```
Comment (apedroferreira) [inline packages/toolpad-app/src/components/HelpTooltipIcon.tsx:6]:
```suggestion
  helpText: React.ReactNode;
```

Actually my bad, looks like `string |` is redundant here and `React.ReactNode` includes strings among other things. I didn't know until I checked a while ago!

**Pull request merged**